[CL-3529] Convert project folder images to React Query

amanda-anderson opened the pull request:
# Changelog
### Fixed
* Fix project image upload
### Technical
* Convert project folder images to React Query

amanda-anderson committed:
WIP Add project folder image endpoints

amanda-anderson committed:
Convert remaining endpoints

amanda-anderson committed:
Add tests

amanda-anderson committed:
Merge branch 'CL-3528-project-folder-files' into cl-3529-project-folder-images-convert

amanda-anderson committed:
Remove deadcode

amanda-anderson committed:
Fix variable name

amanda-anderson committed:
use mutateAsync

amanda-anderson committed:
Remove unnecessary check

amanda-anderson committed:
Merge branch 'master' into cl-3529-project-folder-images-convert

IvaKop commented on front/app/api/project_folder_images/types.ts:
```diff
@@ -1,8 +1,23 @@
+
+export type IGetImagesQueryParameters = {
+  folderId: string;
+  streamParams?: IStreamParams | null;
```
Let's not use streamParams as a namee anymore - you can just add image there, no?

IvaKop commented on front/app/api/project_folder_images/useAddProjectFolderImage.ts:
```diff
@@ -0,0 +1,35 @@
+  fetcher<IProjectFolderImage>({
+    path: `/project_folders/${folderId}/images`,
+    action: 'post',
+    body: { image: { image: base64 } },
```
No need to nest it in this way either, we can just have `image: base64`

amanda-anderson replied:
Hmm I actually get an error from the BE if I use only `{ image: base64 }`. It looks like it's nested for some of the other addImage endpoints as well :thinking: 

Error:
![image](https://github.com/CitizenLabDotCo/citizenlab/assets/33987955/dcca9039-8f32-462c-9058-1042b24946b2)



E.g. Nested in add initiative image:
```
export interface AddInitiativeImageObject {
  initiativeId: string;
  image: {
    image: string;
  };
}
```

IvaKop replied:
Ok, in that case let's leave it as is to keep it consistent with the rest of the image hooks 👍

IvaKop reviewed (approved):
Looks good - some minor comments to address. Also, I don't see the service in the deleted files - should we delete it?

amanda-anderson commented:
@IvaKop Ah - it seems I just moved the file/renamed it as the types.ts and removed what I didn't need :) But yes - the original services file is gone! :+1: 

![image](https://github.com/CitizenLabDotCo/citizenlab/assets/33987955/58f4b8d9-6aa5-4539-9c52-00f2f1b7f1b1)

amanda-anderson committed:
Fix get

IvaKop commented:
@amanda-anderson There seems to be some dead code detected but other than that this one is good to go 👍

amanda-anderson committed:
Merge branch 'master' into cl-3529-project-folder-images-convert

amanda-anderson committed:
Remove deadcode

amanda-anderson committed:
Fix bug with project card image

amanda-anderson committed:
Update tests
